### ISSUE TITLE
FR: Analytics support on watchOS

### ISSUE THREAD

**Issue by jroets** [firebase/firebase-ios-sdk#4557]
Please add Firebase Analytics support to SDK on watchOS, so that we can record to Firebase Analytics directly from watch apps in the same way we can on iOS apps.

To my knowledge right now, to record analytics data from a watch app to Firebase Analytics requires proxying through the SDK running on the iPhone. Or else don't use Firebase for analytics and instead record to GA using its web API (there is no GA SDK for watchOS). Or use a non-Google analytics provider that actually does offer a watchOS SDK, like Yahoo's Flurry SDK. Less than ideal for those of us who want to use Firebase Analytics from a watch app.

**Comment by hannesjung** [comment]
+1

**Comment by nesterman** [comment]
+1

**Comment by jostster** [comment]
+1, Is analytics for iOS open source? I'm not seeing the code for doing community support for watchOS.

**Comment by paulb777** [comment]
@jostster FirebaseAnalytics is not open source.

**Comment by jostster** [comment]
@paulb777 Do you know if it is on the roadmap to allow it to work on watchOS? It seems https://github.com/firebase/firebase-ios-sdk/issues/4558 will be held up by the community without the watchOS support on Analytics since it looks like the FirebaseCrashlytics backend activation requires the Analytics to ping the server.

**Comment by paulb777** [comment]
@jostster I can't speak about roadmaps, but I can say that watchOS is a lower priority than Catalyst, tvOS, and macOS.

Also, Crashlytics does have support for Catalyst, even thought Analytics is not available.

**Comment by kutsin** [comment]
<img width="442" alt="Screenshot1" src="https://user-images.githubusercontent.com/14885155/93630755-29b02400-f9f3-11ea-9e2e-35a368b75321.png">

Any updates on this?

**Comment by paulb777** [comment]
@kutsin Nothing new to report about Analytics platform coverage.

We'll eliminate that warning in an upcoming release in the meantime.

**Comment by SwiftiSwift** [comment]
Still no watchOS support for analytics?

**Comment by paulb777** [comment]
Not yet. We introduced macOS, catalyst and tvOS in 8.9.0 and will try to follow up with watchOS.

Please keep adding thumbs-up's to this feature request to help us with prioritization.

**Comment by plotkin** [comment]
+1

**Comment by curiousdustin** [comment]
@paulb777, Thank you for the updates on this.

Are you able to tell us the nature of the hold up on this?

Is it simply a matter of prioritization and dev resources?

Or, is there some dependency that needs to be resolved first? Perhaps another related Google library, or some library or feature that Apple needs to add or support within watchOS?

Also, is there any official recommendation from your team on how developers should approach analytics for their watchOS apps when Firebase is already being used for analytics and other features within respective companion iOS apps? @jroets mentioned some alternatives, is one of them recommended?

**Comment by wangjian81** [comment]
+1

**Comment by MateuszW13** [comment]
+1

**Comment by gpfister** [comment]
Hi, 

I face some issue building for watchOS using same code base as the companion iOS app. I think it should be nice to have an "empty" framework, especially that SPM first try to resolve dependencies even if there's a condition.

I would love overall Analytics and Crashlytics on watchOS, but as I understand it can be a challenge, but meanwhile just making the curent package available for watchOS, event if it doesn't build, would allow to using target dependency condition to ignore it for watchOS (similar to building FirebaseFirestore framework, which gives an error (about storage).

Thanks in advance for your support.

Kind regards,
Greg

**Comment by paulb777** [comment]
@gpfister Interesting idea. Would you create a separate issue to track?

**Comment by gpfister** [comment]
@paulb777 I have created the feature request #9966

**Comment by samkulkarni3** [comment]
I would love to see this feature Implemented also.

**Comment by gohnjanotis** [comment]
In the meantime, does anyone have a workaround to get Xcode Previews to work with a watchOS target when Firebase is added to the iOS target?

I can build and run fine on an Apple Watch simulator or real Apple Watch, but when I try to run Xcode Previews I encounter this issue: #7610

That issue is closed, because it seems it was resolved at some point in Xcode 13, but I am experiencing the issue again on Firebase 10.0.0 and Xcode 14.0.1 without any workaround to use Xcode Previews for watchOS apps.

**Comment by gpfister** [comment]
Hi,

@gohnjanotis,  I don't have an easy workaround. I ended up splitting into multiple libraries and use pre-compiler instruction based on the target OS (`#if os(iOS) ... #endif`) in order to avoid compilation issues. It worked, as the code that was calling for analytics only was necessary on iOS, so it was possible to split (for example, I had a AuthLite lib and Auth lib, the AuthLite would have the basic for the watch, while the Auth "full" would have the extension for iOS).

I hope it helps. In any case I would say that the issue is really on the SPM side, as I don't think sub-dependencies should be resolved if the dependency it self is not to be considered...

**Comment by santiagogarzon** [comment]
+1

**Comment by RichardFevrier** [comment]
+1

**Comment by pansuriyaravi** [comment]
+1

**Comment by venusbhatia** [comment]
+1,
Please add the support of Analytics to watchOS and visionOS, too. (if possible)

**Comment by denandreychuk** [comment]
+1. It would be great to have Firebase Analytics on watchOS

**Comment by LOOHP** [comment]
+1. It would be nice to have Firebase Analytics on Standalone WatchOS like their WearOS counterparts.

**Comment by opgeROBt** [comment]
+1

**Comment by rockshassa** [comment]
+1 please @paulb777

**Comment by Flatout73** [comment]
+1

**Comment by GreeMoz** [comment]
+1

**Comment by alishaiduc7** [comment]
+1

**Comment by bpkelsey** [comment]
+1

**Comment by robinsalehjan** [comment]
Would love to have this! Is there a public roadmap or is it possible to contribute in anyway?

**Comment by marcgg** [comment]
+1

**Comment by Jager-yoo** [comment]
Our production iOS app already uses `Firebase Analytics`, so I also need to send logs from the watchOS target to the same place.
But since Firebase Analytics doesn’t officially support watchOS, I’m forced to proxy events through the companion iOS app. You know, this process is very complex and really time consuming.

`Amplitude` supports watchOS officially and is very easy to set up, but changing the whole analytics system isn’t a realistic option. I know there is a community workaround, but it uses `CocoaPods`. As you might know, CocoaPods will become read-only in December 2026, so this solution won’t last long or be sustainable.
Please consider expediting official support for watchOS or even just a lightweight SPM package for watchOS would be incredibly helpful.

**Comment by MoonPandragon** [comment]
+1

**Comment by LeonidKokhnovich** [comment]
+1

**Comment by YannickvanIJsseldijk** [comment]
+1
